x-pack/plugins/ml/public/components/annotations_table/annotations_table.js

```diff
@@ -0,0 +1,342 @@
+
+    const _g = rison.encode({
+      ml: {
+        jobIds: [this.props.jobs[0].job_id]
+      },
+      refreshInterval: {
+        display: 'Off',
+        pause: false,
+        value: 0
+      },
+      time: {
+        from,
+        to,
+        mode: 'absolute'
+      }
+    });
+
+    const appState = {
+      filters: [],
+      query: {
+        query_string: {
+          analyze_wildcard: true,
+          query: '*'
+        }
+      }
+    };
+
+    if (annotation !== undefined) {
+      appState.mlTimeSeriesExplorer = {
+        zoom: {
+          from: new Date(annotation.timestamp).toISOString(),
+          to: new Date(annotation.end_timestamp).toISOString()
+        }
+      };
+    }
+
+    const _a = rison.encode(appState);
+
+    const url = `?_g=${_g}&_a=${_a}`;
+    addItemToRecentlyAccessed('timeseriesexplorer', this.props.jobs[0].job_id, url);
+    window.open(`${chrome.getBasePath()}/app/ml#/timeseriesexplorer${url}`, '_self');
+  }
+
+  onMouseOverRow = (record) => {
+    if (this.mouseOverRecord !== undefined) {
+      if (this.mouseOverRecord.rowId !== record.rowId) {
+        // Mouse is over a different row, fire mouseleave on the previous record.
+        mlAnomaliesTableService.anomalyRecordMouseleave.changed(this.mouseOverRecord, 'annotation');
+
+        // fire mouseenter on the new record.
+        mlAnomaliesTableService.anomalyRecordMouseenter.changed(record, 'annotation');
+      }
+    } else {
+      // Mouse is now over a row, fire mouseenter on the record.
+      mlAnomaliesTableService.anomalyRecordMouseenter.changed(record, 'annotation');
+    }
+
+    this.mouseOverRecord = record;
+  }
+
+  onMouseLeaveRow = () => {
+    if (this.mouseOverRecord !== undefined) {
+      mlAnomaliesTableService.anomalyRecordMouseleave.changed(this.mouseOverRecord, 'annotation');
+      this.mouseOverRecord = undefined;
+    }
+  };
+
+  render() {
+    const {
+      isSingleMetricViewerLinkVisible = true,
+      isNumberBadgeVisible = false
+    } = this.props;
+
+    if (this.props.annotations === undefined) {
+      if (this.state.isLoading === true) {
+        return (
+          <EuiFlexGroup justifyContent="spaceAround">
+            <EuiFlexItem grow={false}><EuiLoadingSpinner size="l"/></EuiFlexItem>
+          </EuiFlexGroup>
+        );
+      }
+
+      if (this.state.errorMessage !== undefined) {
+        return (
+          <EuiCallOut
+            title={this.state.errorMessage}
+            color="danger"
+            iconType="cross"
+          />
+        );
+      }
+    }
+
+    const annotations = this.props.annotations || this.state.annotations;
+
+    if (annotations.length === 0) {
+      return (
+        <EuiCallOut
+          title="No annotations created for this job"
+          iconType="iInCircle"
+        >
+          <p>
+            To create an annotation,
+            open the <EuiLink onClick={() => this.openSingleMetricView()}>Single Metric Viewer</EuiLink>
```

Comment:
I don't think you need to wrap the `this.openSingleMetricView` method here since you aren't passing in anything. Should be able to just do something like `onClick={this.openSingleMetricView}`

Reply:
👍 Done in e251767910.